File: Araki/JPL/ch14/ex14_03/Adder.java
package ch14.ex14_03;

public class Adder {
    private int current_num = 0;
    
    public Adder() {
        current_num = 0;
    }
    
    public synchronized void add(int num) {
        this.current_num = this.current_num + num;
    }
    
    public int getNum() {
        return this.current_num;
    }
}

File: Araki/JPL/ch14/ex14_03/Adder.go
package Adder

import (
	//"fmt"
	)

type Adder struct {
	current_num int
}

func (adder *Adder) Add (num int) {
	adder.current_num = adder.current_num + num
	//fmt.Println(adder.current_num)
}

